### ISSUE TITLE
How to do classification task.

### ISSUE THREAD

**Issue by TonyHuBD** [ma-compbio/Hyper-SAGNN#2]
Hello, Hyper-SAGNN is a really a good work. We want to conduct some experiments using this model. Could you please tell me how to reproduce node classification task in the paper using your code.

**Comment by ruochiz** [comment]
Thanks for you interest. The label information is contained in the ```train_data.npz```  of the data Wordnet and MovieLens.
It can be loaded via code such as

The classification task is designed as using a specific ratio of data as training and the rest as testing. For each ratio, we repeated 10 times to avoid the randomness.

The classifier is chosen as the Logistic Regression in the sklearn package. 
Before fitting the classifier, all the embeddings are normalized with ```StandardScaler()``` in the sklearn package.

Here's a skeleton of the experiment I just described above.

```python
train_data = np.load("../DHNE/data/wordnet/train_data.npz",allow_pickle=True)
label = train_data['labels']
ratio = ...
embedding = np.load(...)
for i in range(10):
    X_train, X1test y_train, y_test = train_test_split(embeddings, label, test_size = ratio, random_state=42)
    normalizer = StandardScaler().fit(train)
    train = normalizer.transform(train)
    test = normalizer.transform(test)
    clf = LogisticRegression()
    clf.fit(train, y_train)
    y_pred = clf.predict(test).reshape((-1, 1))
    y_test = y_test.reshape((-1, 1))
    micro, macro = f1_score(y_test, y_pred, average='micro'),f1_score(y_test, y_pred, average='macro')
```

Note that, these two classification are multi-label and multi-class classification respectively. So, you might want to use ```OneVsRestClassifier()``` to wrap the Logistic Regression when needed.

I hope this answers your question.

**Comment by TonyHuBD** [comment]
Thank you for your reply. It's really useful. But how can I get the embedding from model for each node? Is that the result of hadamard power in the model?

**Comment by ruochiz** [comment]
In the ```main.py``` there's a function called ```save_embeddings()```. It is called automatically I believe. This will produce some files named as ```mymodel_0/1/2_origin.npy``` in the root dir where the number 0 corresponds to the first, second or third node type.

Feel free to change file name or directory in the ```main.py```

**Comment by TonyHuBD** [comment]
Thanks a a lot. It's really helpful.

**Comment by TonyHuBD** [comment]
Sorry, it's me again. I see `save_embeddings( )` is called twice. The first time is after the model building. The second time is in a function called `train` when `only_rw=True`. But when you call the train function you set `only_rw=False`. So the embeddings is saved only after the model building. Is that right?

**Comment by ruochiz** [comment]
You are right. In that case, feel free to call it in each epoch, or just comment out the ```if only_rw:``` line of code. I suppose I accidentally deleted the code to use the function after training when cleaning up the code.

**Comment by TonyHuBD** [comment]
When I using the dataset with only one type of nodes. The model can't be trained correctly.  The parameters of TiedAutoEncoder will be trained to nan. I don't know why. Have you done any experiments in this situation?

**Comment by ruochiz** [comment]
The code in this repo is for reproducing the results for the 4 datasets (heterogeneous 3-uniform hypergraph). If you are applying it to a homogeneous hypergraph, you should change the ```generate_embeddings``` function, as now it would just use the decomposed adjacency matrix as features.
